baseapp/baseapp.go

```diff
@@ -507,8 +509,7 @@ func (app *BaseApp) StoreConsensusParams(ctx sdk.Context, cp *abci.ConsensusPara
 	app.paramStore.Set(ctx, ParamStoreKeyBlockParams, cp.Block)
 	app.paramStore.Set(ctx, ParamStoreKeyEvidenceParams, cp.Evidence)
 	app.paramStore.Set(ctx, ParamStoreKeyValidatorParams, cp.Validator)
-	// We're explicitly not storing the Tendermint app_version in the param store. It's
-	// stored instead in the x/upgrade store, with its own bump logic.
+	app.paramStore.Set(ctx, ParamStoreKeyVersionParams, cp.Version)
```

Comment:
We don't store it in upgrade because we have a different module hence we should store it here. In v0.50.x, the SDK also begins to store the app version in the ParamStore of BaseApp